waspc/data/Generator/templates/react-app/src/index.js

```diff
@@ -32,6 +36,21 @@ async function startApp() {
   serviceWorker.unregister()
 }
 
+// NOTE: Since users will likely have the backend running on a different domain than
+// the frontend, we are unable to set the token:
+// (a) on the page load, as the index.html is not served by Node, nor
+// (b) via a cookie, since the frontend JS will not be able to access a cross-domain cookie.
+async function setCsrfTokenHeader() {
+  try {
+    const token = await api.get(config.apiUrl + '/csrf-token')
+    if (token.data) {
```

Comment:
Yes, I guess this is the same issue of error handling then we can't obtain csrf token.
I am just wondering, in this function, should we handle case of falsy token.data or not.
If we don't, and if token.data is not set (which we are not sure is possible or not - ah dynamic languages :D), then the function setCsrfTokenHeader() will fail silently, which sucks a bit maybe? Maybe it would be nicer to at least know smth went wrong?
But on the other hand if this will just make the whole thing overly complex right now, which it might, then I am ok for ignoring it for now. Requests will fail, stuff will go wrong, and we can figure out in the future how to properly handle all this when we are a bit smarter.

Reply:
Yeah, I think we may want to open an issue to think on this a bit more. Or we can have a meeting tomorrow maybe? I'm not sure offhand what the best approach would even be here... should every future request be queued until we get it? Or just failed? Should we notify the user? If so, how?